[Bug]: parser error Couldn't parse parameter override rule: '--param robot_description

### What happened?

I'm trying to load a locobot in gazebo with this command:
```
ros2 launch interbotix_xslocobot_sim xslocobot_gz_classic.launch.py robot_model:=locobot_px100 base_type:=create3
```
I get the following error:
```
asaoines@ros2-humble-vm:/locobot/workspace$ ros2 launch interbotix_xslocobot_sim xslocobot_gz_classic.launch.py robot_model:=locobot_px100 base_type:=create3
[INFO] [launch]: All log files can be found below /home/asaoines/.ros/log/2024-11-08-22-30-20-623563-ros2-humble-vm-48567
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [gzserver-1]: process started with pid [48569]
[INFO] [gzclient-2]: process started with pid [48571]
[INFO] [spawn_entity.py-3]: process started with pid [48573]
[INFO] [robot_state_publisher-4]: process started with pid [48575]
[INFO] [rviz2-5]: process started with pid [48577]
[gzserver-1] xcb_connection_has_error() returned true
[gzserver-1] xcb_connection_has_error() returned true
[ERROR] [rviz2-5]: process has died [pid 48577, exit code -6, cmd '/opt/ros/humble/lib/rviz2/rviz2 -d /locobot/workspace/install/interbotix_xslocobot_sim/share/interbotix_xslocobot_sim/rviz/xslocobot_gz_classic.rviz -f locobot/base_link --ros-args -r __node:=rviz2 -r __ns:=/locobot --params-file /tmp/launch_params_v54x4flz'].
[gzserver-1] [INFO] [1731105023.827951106] [gazebo_ros2_control]: Loading gazebo_ros2_control plugin
[gzserver-1] [INFO] [1731105023.833508752] [locobot.gazebo_ros2_control]: Starting gazebo_ros2_control plugin in namespace: /locobot
[gzserver-1] [INFO] [1731105023.833818578] [locobot.gazebo_ros2_control]: Starting gazebo_ros2_control plugin in ros 2 node: gazebo_ros2_control
[gzserver-1] [INFO] [1731105023.838555252] [locobot.gazebo_ros2_control]: connected to service!! robot_state_publisher
[gzserver-1] [INFO] [1731105023.839792022] [locobot.gazebo_ros2_control]: Received urdf from param server, parsing...
[gzserver-1] [INFO] [1731105023.839864951] [locobot.gazebo_ros2_control]: Loading parameter files /locobot/workspace/install/interbotix_xslocobot_sim/share/interbotix_xslocobot_sim/config/create3_version/locobot_px100_controllers.yaml
[gzserver-1] [rcutils|error_handling.c:65] an error string (message, file name, or formatted message) will be truncated
[gzserver-1] [ERROR] [1731105023.841246228] [locobot.gazebo_ros2_control]: parser error Couldn't parse parameter override rule: '--param robot_description:=<?xml version="1.0" ?>
[gzserver-1] <!-- =================================================================================== -->
[gzserver-1] <!-- |    This document was autogenerated by xacro from /locobot/workspace/install/interbotix_xslocobot_descriptions/share/interbotix_xslocobot_descriptions/urdf/locobot.urdf.xacro | -->
[gzserver-1] <!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
[gzserver-1] <!-- =================================================================================== -->
[gzserver-1] <robot name="locobot">
[gzserver-1]   <material name="interbotix_black">
[gzserver-1]     <texture filename="package://interbotix_common_sim/media/materials/textures/interbotix_black.png"/>
[gzserver-1]   </material>
[gzserver-1]   <!-- kobuki -->
[gzserver-1]   <!-- Much of, at ./src/rcl/arguments.c:343
[gzserver-1] 
[INFO] [spawn_entity.py-3]: process has finished cleanly [pid 48573]
```


### Robot Model

locobot_px100 but with gazebo so it's a bit moot

### Operating System

Ubuntu 22.04

### ROS Distro

ROS 2 Humble

### Steps To Reproduce

_No response_

### Relevant log output

_No response_

### Additional Info

It works if I edit DeclareInterbotixXSLoCoBotRobotDescriptionLaunchArgument in xs_launch.py to have the default robot_description be sourced from a URDF file (via Command(['cat'...])) that I generated with this command, which strips out comments from the URDF:

```
sh -c "/opt/ros/humble/bin/xacro /locobot/workspace/install/interbotix_xslocobot_descriptions/share/interbotix_xslocobot_descriptions/urdf/locobot.urdf.xacro arm_model:=mobile_px100 robot_name:=locobot base_model:=create3 robot_model:=locobot_px100 robot_name:=locobot use_gripper:=true show_ar_tag:=true show_gripper_bar:=true show_gripper_fingers:=true use_lidar:=false external_urdf_loc:= hardware_type:=gz_classic" | sed -e '/<\!--/,/-->/d'
```

I tried to edit that function to automatically strip out comments after generating the URDF from Xacro but I couldn't get it to work.

This approach was inspired by https://github.com/ros-controls/gazebo_ros2_control/issues/295

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_launch/xs_launch.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_launch/xs_launch.py
@@ -99,11 +99,11 @@ class DeclareInterbotixXSArmRobotDescriptionLaunchArgument(DeclareLaunchArgument
 
 class DeclareInterbotixXSLoCoBotRobotDescriptionLaunchArgument(DeclareLaunchArgument):
     """Generate a URDF of a LoCoBot through a modified DeclareLaunchArgument object."""
-
     def __init__(
         self,
         *,
         default_value: Optional[SomeSubstitutionsType] = Command([
+            'sh -c "',
             FindExecutable(name='xacro'), ' ',
             PathJoinSubstitution([
                 FindPackageShare('interbotix_xslocobot_descriptions'),
@@ -122,6 +122,9 @@ class DeclareInterbotixXSLoCoBotRobotDescriptionLaunchArgument(DeclareLaunchArgu
             'use_lidar:=', LaunchConfiguration('use_lidar'), ' ',
             'external_urdf_loc:=', LaunchConfiguration('external_urdf_loc'), ' ',
             'hardware_type:=', LaunchConfiguration('hardware_type'), ' ',
+            # Stripping comments from the URDF is necessary for gazebo_ros2_control to parse the
+            # robot_description parameter override
+            '| ', FindExecutable(name='perl'), ' -0777 -pe \'s/<!--.*?-->//gs\'"'
         ]),
         **kwargs
     ) -> None:

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_launch/xs_launch.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_launch/xs_launch.py
@@ -99,6 +99,7 @@ class DeclareInterbotixXSArmRobotDescriptionLaunchArgument(DeclareLaunchArgument
 
 class DeclareInterbotixXSLoCoBotRobotDescriptionLaunchArgument(DeclareLaunchArgument):
     """Generate a URDF of a LoCoBot through a modified DeclareLaunchArgument object."""
+
     def __init__(
         self,
         *,

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/create3.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/create3.py
@@ -33,13 +33,13 @@ These classes can be used to control a Create 3 mobile base for Interbotix X-Ser
 Python.
 """
 
-from builtin_interfaces.msg import Duration
 from geometry_msgs.msg import Pose
 from interbotix_xs_modules.xs_robot.core import InterbotixRobotXSCore
 from interbotix_xs_modules.xs_robot.mobile_base import InterbotixMobileBaseInterface
 from irobot_create_msgs.msg import AudioNote, AudioNoteVector
 from irobot_create_msgs.srv import ResetPose
 from rclpy.constants import S_TO_NS
+from rclpy.duration import Duration
 from std_msgs.msg import Header
 
 
@@ -78,28 +78,28 @@ class InterbotixCreate3Interface(InterbotixMobileBaseInterface):
             nav_timeout_sec=nav_timeout_sec,
             use_nav=use_nav,
         )
-        self.pub_base_sound = self.core.create_publisher(
+        self.pub_base_sound = self.core.get_node().create_publisher(
             msg_type=AudioNoteVector,
             topic='mobile_base/cmd_audio',
             qos_profile=1
         )
-        self.client_reset_pose = self.core.create_client(
+        self.client_reset_pose = self.core.get_node().create_client(
             srv_type=ResetPose,
             srv_name='mobile_base/reset_pose',
         )
 
-        self.core.get_clock().sleep_for(Duration(nanosec=int(0.5*S_TO_NS)))
-        self.core.get_logger().info('Initialized InterbotixCreate3Interface!')
+        self.core.get_node().get_clock().sleep_for(Duration(nanoseconds=int(0.5*S_TO_NS)))
+        self.core.get_node().loginfo('Initialized InterbotixCreate3Interface!')
 
     def reset_odom(self) -> None:
         """Reset odometry to zero."""
         future_reset_odom = self.client_reset_pose.call_async(ResetPose.Request(pose=Pose()))
-        self.core.robot_spin_once_until_future_complete(future_reset_odom)
+        self.core.get_node().wait_until_future_complete(future_reset_odom)
         self.play_sound(
             sound=AudioNoteVector(
                 header=Header(
                     frame_id=f'{self.robot_name}/base_link',
-                    stamp=self.core.get_clock().now().to_msg()
+                    stamp=self.core.get_node().get_clock().now().to_msg()
                 ),
                 notes=[AudioNote(frequency=6000, max_runtime=Duration(seconds=1.0))],
                 append=False,

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/locobot.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/locobot.py
@@ -33,15 +33,13 @@ These classes can be used to control a Interbotix X-Series LoCoBots using Python
 """
 
 from enum import Enum
-from threading import Thread
-import time
 
+from interbotix_common_modules.common_robot.robot import InterbotixRobotNode
 from interbotix_xs_modules.xs_robot.arm import InterbotixArmXSInterface
 from interbotix_xs_modules.xs_robot.core import InterbotixRobotXSCore
 from interbotix_xs_modules.xs_robot.gripper import InterbotixGripperXSInterface
 from interbotix_xs_modules.xs_robot.turret import InterbotixTurretXSInterface
 import rclpy
-from rclpy.executors import MultiThreadedExecutor
 from rclpy.logging import LoggingSeverity
 
 
@@ -70,9 +68,9 @@ class InterbotixLocobotXS:
         topic_dxl_joint_states: str = 'dynamixel/joint_states',
         topic_base_joint_states: str = 'mobile_base/joint_states',
         use_nav: bool = False,
-        logging_level: LoggingSeverity = LoggingSeverity.INFO,
+        logging_level: LoggingSeverity = LoggingSeverity.DEBUG,
         node_name: str = 'robot_manipulation',
-        start_on_init: bool = True,
+        node: InterbotixRobotNode = None,
         args=None,
     ):
         """
@@ -105,10 +103,9 @@ class InterbotixLocobotXS:
             ERROR, or FATAL. defaults to INFO
         :param node_name: (optional) name to give to the core started by this class, defaults to
             'robot_manipulation'
-        :param start_on_init: (optional) set to `True` to start running the spin thread after the
-            object is built; set to `False` if intending to sub-class this. If set to `False`,
-            either call the `start()` method later on, or add the core to an executor in another
-            thread.
+        :param node: (optional) `rclpy.Node` or derived class to base this robot's ROS-related
+            components on.  If nothing is given, this robot will create its own node. defaults to
+            `None`.
         """
         self.core = InterbotixRobotXSCore(
             robot_model=robot_model,
@@ -116,6 +113,7 @@ class InterbotixLocobotXS:
             topic_joint_states=topic_dxl_joint_states,
             logging_level=logging_level,
             node_name=node_name,
+            node=node,
             args=args
         )
         self.camera = InterbotixTurretXSInterface(
@@ -162,23 +160,6 @@ class InterbotixLocobotXS:
                     init_node=False
                 )
 
-        if start_on_init:
-            self.start()
-
-    def start(self) -> None:
-        """Start a background thread that builds and spins an executor."""
-        self._execution_thread = Thread(target=self.run)
-        self._execution_thread.start()
-
-    def run(self) -> None:
-        """Thread target."""
-        self.ex = MultiThreadedExecutor()
-        self.ex.add_node(self.core)
-        self.ex.spin()
-
-    def shutdown(self) -> None:
-        """Destroy the node and shut down all threads and processes."""
-        self.core.destroy_node()
-        rclpy.shutdown()
-        self._execution_thread.join()
-        time.sleep(0.5)
+    def get_node(self) -> InterbotixRobotNode:
+        """Return the core robot node for this robot."""
+        return self.core.robot_node

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/locobot.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/locobot.py
@@ -39,7 +39,6 @@ from interbotix_xs_modules.xs_robot.arm import InterbotixArmXSInterface
 from interbotix_xs_modules.xs_robot.core import InterbotixRobotXSCore
 from interbotix_xs_modules.xs_robot.gripper import InterbotixGripperXSInterface
 from interbotix_xs_modules.xs_robot.turret import InterbotixTurretXSInterface
-import rclpy
 from rclpy.logging import LoggingSeverity
 
 

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/locobot.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/locobot.py
@@ -68,7 +68,7 @@ class InterbotixLocobotXS:
         topic_dxl_joint_states: str = 'dynamixel/joint_states',
         topic_base_joint_states: str = 'mobile_base/joint_states',
         use_nav: bool = False,
-        logging_level: LoggingSeverity = LoggingSeverity.DEBUG,
+        logging_level: LoggingSeverity = LoggingSeverity.INFO,
         node_name: str = 'robot_manipulation',
         node: InterbotixRobotNode = None,
         args=None,

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/mobile_base.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/mobile_base.py
@@ -83,20 +83,20 @@ class InterbotixMobileBaseInterface(ABC):
 
         cb_group_mobile_base = ReentrantCallbackGroup()
 
-        self.pub_base_twist = self.core.create_publisher(
+        self.pub_base_twist = self.core.get_node().create_publisher(
             msg_type=Twist,
             topic=topic_cmd_vel,
             qos_profile=1,
             callback_group=cb_group_mobile_base,
         )
-        self.sub_base_states = self.core.create_subscription(
+        self.sub_base_states = self.core.get_node().create_subscription(
             msg_type=JointState,
             topic=topic_base_joint_states,
             callback=self._base_states_cb,
             qos_profile=1,
             callback_group=cb_group_mobile_base,
         )
-        self.sub_base_odom = self.core.create_subscription(
+        self.sub_base_odom = self.core.get_node().create_subscription(
             msg_type=Odometry,
             topic='/mobile_base/odom',
             callback=self._base_odom_cb,
@@ -104,13 +104,13 @@ class InterbotixMobileBaseInterface(ABC):
             callback_group=cb_group_mobile_base,
         )
         self.client_base_nav_to_pose = ActionClient(
-            node=self.core,
+            node=self.core.get_node(),
             action_type=NavigateToPose,
             action_name='navigate_to_pose',
             callback_group=cb_group_mobile_base,
         )
 
-        self.core.get_logger().info('Initialized InterbotixMobileBaseInterface!')
+        self.core.get_node().loginfo('Initialized InterbotixMobileBaseInterface!')
 
     def command_velocity_xyaw(
         self,
@@ -200,7 +200,7 @@ class InterbotixMobileBaseInterface(ABC):
         :details: note that if 'blocking' is `False`, the function will always return `True`
         """
         if not self.use_nav:
-            self.core.get_logger().error('`use_nav` set to `False`. Will not execute navigation.')
+            self.core.get_node().logerror('`use_nav` set to `False`. Will not execute navigation.')
             return False
         goal = NavigateToPose.Goal(
             pose=self._stamp_pose(pose=goal_pose, frame_id=frame_id),
@@ -212,11 +212,11 @@ class InterbotixMobileBaseInterface(ABC):
             feedback_callback=self._nav_to_pose_feedback_cb,
         )
 
-        self.core.robot_spin_once_until_future_complete(future_send_nav_to_pose_goal)
+        self.core.get_node().wait_until_future_complete(future_send_nav_to_pose_goal)
         self.goal_handle = future_send_nav_to_pose_goal.result()
 
         if not self.goal_handle.accepted:
-            self.core.get_logger().error(
+            self.core.get_node().logerror(
                 f'Navigation goal [{goal_pose.position.x}, {goal_pose.position.y}] was rejected.'
             )
             return False
@@ -225,12 +225,12 @@ class InterbotixMobileBaseInterface(ABC):
             while not self.is_nav_complete():
                 fb = self.get_nav_to_pose_feedback()
                 if Duration.from_msg(fb.navigation_time > Duration(seconds=self.nav_timeout_sec)):
-                    self.core.get_logger().error(
+                    self.core.get_node().logerror(
                         f'Navigation time ({fb.navigation_time}) exceeds timeout '
                         f'({self.nav_timeout_sec}). Cancelling navigation goal.'
                     )
                     future_cancel_nav_to_pose_goal = self.goal_handle.cancel_goal_async()
-                    self.core.robot_spin_once_until_future_complete(
+                    self.core.get_node().wait_until_future_complete(
                         future=future_cancel_nav_to_pose_goal
                     )
                 return False
@@ -290,11 +290,11 @@ class InterbotixMobileBaseInterface(ABC):
         """
         if not self.future_nav:
             return True
-        self.core.robot_spin_once_until_future_complete(future=self.future_nav, timeout_sec=0.1)
+        self.core.get_node().wait_until_future_complete(future=self.future_nav, timeout_sec=0.1)
         if self.future_nav.result():
             self.nav_status = self.future_nav.result().status
             if self.nav_status != GoalStatus.STATUS_SUCCEEDED:
-                self.core.get_logger().error(f"Navigation failed with status '{self.nav_status}'.")
+                self.core.get_node().logerror(f"Navigation failed with status '{self.nav_status}'.")
                 return True
         else:
             return False

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/mobile_base.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/mobile_base.py
@@ -294,7 +294,9 @@ class InterbotixMobileBaseInterface(ABC):
         if self.future_nav.result():
             self.nav_status = self.future_nav.result().status
             if self.nav_status != GoalStatus.STATUS_SUCCEEDED:
-                self.core.get_node().logerror(f"Navigation failed with status '{self.nav_status}'.")
+                self.core.get_node().logerror(
+                    f"Navigation failed with status '{self.nav_status}'."
+                )
                 return True
         else:
             return False

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/turret.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/turret.py
@@ -104,6 +104,7 @@ class InterbotixTurretXS:
             logging_level=logging_level,
             node_name=node_name,
         )
+
         self.turret = InterbotixTurretXSInterface(
             core=self.core,
             turret_name=turret_name,
@@ -126,13 +127,13 @@ class InterbotixTurretXS:
     def run(self) -> None:
         """Thread target."""
         self.ex = MultiThreadedExecutor()
-        self.ex.add_node(self.core)
+        self.ex.add_node(self.core.get_node())
         while rclpy.ok():
             self.ex.spin()
 
     def shutdown(self) -> None:
         """Destroy the node and shut down all threads and processes."""
-        self.core.destroy_node()
+        self.core.get_node().destroy_node()
         rclpy.shutdown()
         self._execution_thread.join()
         time.sleep(0.5)
@@ -182,14 +183,16 @@ class InterbotixTurretXSInterface:
         )
 
         while rclpy.ok() and not self.future_group_info.done():
-            rclpy.spin_until_future_complete(self.core, self.future_group_info)
-            rclpy.spin_once(self.core)
+            rclpy.spin_until_future_complete(self.core.get_node(), self.future_group_info)
+            rclpy.spin_once(self.core.get_node())
 
         group_info: RobotInfo.Response = self.future_group_info.result()
         self.pan_name = group_info.joint_names[0]
         self.tilt_name = group_info.joint_names[1]
-        pan_limits = [group_info.joint_lower_limits[0], group_info.joint_upper_limits[0]]
-        tilt_limits = [group_info.joint_lower_limits[1], group_info.joint_upper_limits[1]]
+        pan_limits = [group_info.joint_lower_limits[0],
+                      group_info.joint_upper_limits[0]]
+        tilt_limits = [group_info.joint_lower_limits[1],
+                       group_info.joint_upper_limits[1]]
         pan_position = self.core.joint_states.position[self.core.js_index_map[self.pan_name]]
         tilt_position = self.core.joint_states.position[self.core.js_index_map[self.tilt_name]]
         self.info = {
@@ -222,7 +225,7 @@ class InterbotixTurretXSInterface:
             tilt_profile_velocity,
             tilt_profile_acceleration
         )
-        self.core.get_logger().info((
+        self.core.get_node().loginfo((
             f'Turret Group Name: {turret_name}\n'
             f'Pan Name: {self.pan_name}, '
             f'Profile Type: {pan_profile_type}, '
@@ -233,7 +236,7 @@ class InterbotixTurretXSInterface:
             f'Profile Velocity: {tilt_profile_velocity:.1f}, '
             f'Profile Acceleration: {tilt_profile_acceleration:.1f}'
         ))
-        self.core.get_logger().info('Initialized InterbotixTurretXSInterface!')
+        self.core.get_node().loginfo('Initialized InterbotixTurretXSInterface!')
 
     def set_trajectory_profile(
         self,
@@ -267,7 +270,7 @@ class InterbotixTurretXSInterface:
                         value=profile_velocity
                     )
                 )
-                self.core.robot_spin_once_until_future_complete(
+                self.core.get_node().executor.spin_once_until_future_complete(
                     future=future_profile_velocity,
                     timeout_sec=0.1
                 )
@@ -280,14 +283,15 @@ class InterbotixTurretXSInterface:
                         value=int(profile_velocity * 1000)
                     )
                 )
-                self.core.robot_spin_once_until_future_complete(
+                self.core.get_node().executor.spin_once_until_future_complete(
                     future=future_profile_velocity,
                     timeout_sec=0.1
                 )
             self.info[joint_name]['profile_velocity'] = profile_velocity
         if (
             (profile_acceleration is not None) and
-            (profile_acceleration != self.info[joint_name]['profile_acceleration'])
+            (profile_acceleration !=
+             self.info[joint_name]['profile_acceleration'])
         ):
             if (self.info[joint_name]['profile_type'] == 'velocity'):
                 future_profile_acceleration = self.core.srv_set_reg.call_async(
@@ -298,7 +302,7 @@ class InterbotixTurretXSInterface:
                         value=profile_acceleration
                     )
                 )
-                self.core.robot_spin_once_until_future_complete(
+                self.core.get_node().executor.spin_once_until_future_complete(
                     future=future_profile_acceleration,
                     timeout_sec=0.1
                 )
@@ -311,7 +315,7 @@ class InterbotixTurretXSInterface:
                         value=int(profile_acceleration * 1000)
                     )
                 )
-                self.core.robot_spin_once_until_future_complete(
+                self.core.get_node().executor.spin_once_until_future_complete(
                     future=future_profile_acceleration,
                     timeout_sec=0.1
                 )
@@ -349,18 +353,21 @@ class InterbotixTurretXSInterface:
             (self.info[joint_name]['lower_limit'] <= position) and
             (position <= self.info[joint_name]['upper_limit'])
         ):
-            self.set_trajectory_profile(joint_name, profile_velocity, profile_acceleration)
-            self.core.pub_single.publish(JointSingleCommand(name=joint_name, cmd=position))
+            self.set_trajectory_profile(
+                joint_name, profile_velocity, profile_acceleration)
+            self.core.pub_single.publish(
+                JointSingleCommand(name=joint_name, cmd=position))
             self.info[joint_name]['command'] = position
             if (self.info[joint_name]['profile_type'] == 'time' and blocking):
                 print(self.info[joint_name]['profile_velocity'])
-                self.core.get_clock().sleep_for(
-                   Duration(nanoseconds=int(self.info[joint_name]['profile_velocity'] * S_TO_NS))
+                self.core.get_node().get_clock().sleep_for(
+                    Duration(nanoseconds=int(
+                        self.info[joint_name]['profile_velocity'] * S_TO_NS))
                 )
             else:
-                self.core.get_clock().sleep_for(Duration(nanoseconds=int(delay * S_TO_NS)))
+                self.core.get_node().get_clock().sleep_for(Duration(nanoseconds=int(delay * S_TO_NS)))
         else:
-            self.core.get_logger().error(
+            self.core.get_node().logerror(
                 f"Goal position is outside the '{joint_name}' joint's limits. Will not execute."
             )
 
@@ -527,12 +534,12 @@ class InterbotixTurretXSInterface:
                 pan_profile_acceleration
             )
             self.set_trajectory_profile(
-               self.tilt_name,
-               tilt_profile_velocity,
-               tilt_profile_acceleration
+                self.tilt_name,
+                tilt_profile_velocity,
+                tilt_profile_acceleration
             )
             self.core.pub_group.publish(
-               JointGroupCommand(name=self.turret_name, cmd=[pan_position, tilt_position]))
+                JointGroupCommand(name=self.turret_name, cmd=[pan_position, tilt_position]))
             self.info[self.pan_name]['command'] = pan_position
             self.info[self.tilt_name]['command'] = tilt_position
             if (
@@ -540,7 +547,7 @@ class InterbotixTurretXSInterface:
                 (self.info[self.tilt_name]['profile_type'] == 'time') and
                 (blocking)
             ):
-                self.core.get_clock().sleep_for(
+                self.core.get_node().get_clock().sleep_for(
                     Duration(
                         nanoseconds=int(max(
                             self.info[self.pan_name]['profile_velocity'],
@@ -549,9 +556,9 @@ class InterbotixTurretXSInterface:
                     )
                 )
             else:
-                self.core.get_clock().sleep_for(Duration(nanoseconds=int(delay * S_TO_NS)))
+                self.core.get_node().get_clock().sleep_for(Duration(nanoseconds=int(delay * S_TO_NS)))
         else:
-            self.core.get_logger().error(
+            self.core.get_node().logerror(
                 'One or both goal positions are outside the limits. Will not execute'
             )
 
@@ -584,7 +591,7 @@ class InterbotixTurretXSInterface:
                     profile_acceleration=profile_acceleration
                 )
             )
-            self.core.robot_spin_once_until_future_complete(
+            self.core.get_node().executor.spin_once_until_future_complete(
                 future=future_operating_modes,
                 timeout_sec=0.1
             )
@@ -601,7 +608,7 @@ class InterbotixTurretXSInterface:
                     profile_acceleration=int(profile_acceleration * 1000)
                 )
             )
-            self.core.robot_spin_once_until_future_complete(
+            self.core.get_node().executor.spin_once_until_future_complete(
                 future=future_operating_modes,
                 timeout_sec=0.1
             )

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/turret.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/turret.py
@@ -360,7 +360,9 @@ class InterbotixTurretXSInterface:
                         self.info[joint_name]['profile_velocity'] * S_TO_NS))
                 )
             else:
-                self.core.get_node().get_clock().sleep_for(Duration(nanoseconds=int(delay * S_TO_NS)))
+                self.core.get_node().get_clock().sleep_for(
+                    Duration(nanoseconds=int(delay * S_TO_NS))
+                )
         else:
             self.core.get_node().logerror(
                 f"Goal position is outside the '{joint_name}' joint's limits. Will not execute."
@@ -551,7 +553,9 @@ class InterbotixTurretXSInterface:
                     )
                 )
             else:
-                self.core.get_node().get_clock().sleep_for(Duration(nanoseconds=int(delay * S_TO_NS)))
+                self.core.get_node().get_clock().sleep_for(
+                    Duration(nanoseconds=int(delay * S_TO_NS))
+                )
         else:
             self.core.get_node().logerror(
                 'One or both goal positions are outside the limits. Will not execute'

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/turret.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/turret.py
@@ -189,8 +189,8 @@ class InterbotixTurretXSInterface:
         group_info: RobotInfo.Response = self.future_group_info.result()
         self.pan_name = group_info.joint_names[0]
         self.tilt_name = group_info.joint_names[1]
-        pan_limits = [group_info.joint_lower_limits[0],                       group_info.joint_upper_limits[0]]
-        tilt_limits = [group_info.joint_lower_limits[1],                        group_info.joint_upper_limits[1]]
+        pan_limits = [group_info.joint_lower_limits[0], group_info.joint_upper_limits[0]]
+        tilt_limits = [group_info.joint_lower_limits[1], group_info.joint_upper_limits[1]]
         pan_position = self.core.joint_states.position[self.core.js_index_map[self.pan_name]]
         tilt_position = self.core.joint_states.position[self.core.js_index_map[self.tilt_name]]
         self.info = {

--- a/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/turret.py
+++ b/interbotix_xs_toolbox/interbotix_xs_modules/interbotix_xs_modules/xs_robot/turret.py
@@ -189,10 +189,8 @@ class InterbotixTurretXSInterface:
         group_info: RobotInfo.Response = self.future_group_info.result()
         self.pan_name = group_info.joint_names[0]
         self.tilt_name = group_info.joint_names[1]
-        pan_limits = [group_info.joint_lower_limits[0],
-                      group_info.joint_upper_limits[0]]
-        tilt_limits = [group_info.joint_lower_limits[1],
-                       group_info.joint_upper_limits[1]]
+        pan_limits = [group_info.joint_lower_limits[0],                       group_info.joint_upper_limits[0]]
+        tilt_limits = [group_info.joint_lower_limits[1],                        group_info.joint_upper_limits[1]]
         pan_position = self.core.joint_states.position[self.core.js_index_map[self.pan_name]]
         tilt_position = self.core.joint_states.position[self.core.js_index_map[self.tilt_name]]
         self.info = {
@@ -290,8 +288,7 @@ class InterbotixTurretXSInterface:
             self.info[joint_name]['profile_velocity'] = profile_velocity
         if (
             (profile_acceleration is not None) and
-            (profile_acceleration !=
-             self.info[joint_name]['profile_acceleration'])
+            (profile_acceleration != self.info[joint_name]['profile_acceleration'])
         ):
             if (self.info[joint_name]['profile_type'] == 'velocity'):
                 future_profile_acceleration = self.core.srv_set_reg.call_async(
@@ -353,10 +350,8 @@ class InterbotixTurretXSInterface:
             (self.info[joint_name]['lower_limit'] <= position) and
             (position <= self.info[joint_name]['upper_limit'])
         ):
-            self.set_trajectory_profile(
-                joint_name, profile_velocity, profile_acceleration)
-            self.core.pub_single.publish(
-                JointSingleCommand(name=joint_name, cmd=position))
+            self.set_trajectory_profile(joint_name, profile_velocity, profile_acceleration)
+            self.core.pub_single.publish(JointSingleCommand(name=joint_name, cmd=position))
             self.info[joint_name]['command'] = position
             if (self.info[joint_name]['profile_type'] == 'time' and blocking):
                 print(self.info[joint_name]['profile_velocity'])

